### ISSUE TITLE
OpenMP openmp.range on Develop Runs Forever on POWER7+ with RHEL7 and GCC4.8.5

### ISSUE THREAD

**Issue by nmhamster** [kokkos/kokkos#995]
Compiled on POWER7+ (Big Endian) with GCC 4.8.5. Test start to execute and then hang running forever.

```
[==========] Running 83 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 83 tests from openmp
[ RUN      ] openmp.init
[       OK ] openmp.init (0 ms)
[ RUN      ] openmp.dispatch
[       OK ] openmp.dispatch (225 ms)
[ RUN      ] openmp.compiler_macros
[       OK ] openmp.compiler_macros (1 ms)
[ RUN      ] openmp.policy_construction
[       OK ] openmp.policy_construction (0 ms)
[ RUN      ] openmp.impl_shared_alloc
[       OK ] openmp.impl_shared_alloc (353 ms)
[ RUN      ] openmp.range_for
```

**Comment by crtrott** [comment]
Might be related to issue #1013.

**Comment by ibaned** [comment]
does this still happen?

**Comment by nmhamster** [comment]
@ibaned - checking, will get back to you.

**Comment by dsunder** [comment]
Fixed by #1240
